Fix: Functional and Unit Tests

SmartArray opened the pull request:
### This PR fixes the remaining issues of the DigiByte Core functional tests and unit tests.

With these changes, the complete test suite SHOULD be complete.

Tested on Ubuntu and Mac OS Monterey

#### Notes

When running the functional tests, make sure you quit all potential listening applications. Sometimes it is still needed to use `./test_runner.py -j1`. At least on my machine there was a port clash on port 18543, for whatever reason.

GreenParticle committed:
Added Gitter

ycagel committed:
Merge pull request #74 from GreenParticle/feature/8.22.0

URL fix: https://dgbwiki.com/

Commit:
Remove odocrypt deployment definition from chainparams.cpp (again).
This is not required if the deployment has been buried.
Additionally it causes a segfault if digibyte was launched with a new data directory.

Commit:
Ignore any 'unknown new rules' message if triggered by a bit within block algo version range

JaredTate committed:
Fix "Unknown new rules activated" Error & Uninitialized Variable nMinerConfirmationWindow, Rever Changes

JaredTate committed:
Add -disabledandelion back

ycagel committed:
Merge pull request #76 from JaredTate/feature/8.22.0

Fix "Unknown new rules activated" Error & Uninitialized Variable nMinerConfirmationWindow, Revert Changes

Commit:
strip dandelion out for complete refactoring

Commit:
strip dandelion out of tests for refactoring

JaredTate committed:
Merge pull request #79 from barrystyle/debug/remove-dandelion

Dandelion removal

Commit:
new legacy implementation of dandelion
complete inv/hash rewrite with dandelion queues

Signed-off-by: barrystyle <baz@tiltpool.com>

Commit:
recommit dandelion tests

sipa committed:
Add pure Python RIPEMD-160

sipa committed:
Swap out hashlib.ripemd160 for own implementation

Commit:
add locks to appropriate functions

Commit:
Fix: Functional: Removed comma at the end of line to make functional tests runnable

Commit:
Refactor: test_node.py: Removed unused code and remerged upstream code

Commit:
Remerge upstream: Functional Tests

Commit:
Fix: RPC (generatetoaddress): Wrong Parameter Index specified.

Commit:
Merge: Replaced Bitcoin with DigiByte

Commit:
Fix: Functional: Generated KeyPairs for 12 nodes in testsuite

Commit:
Dandelion: Added Command Line option -dandelion

Commit:
Fix-Test: Disable Dandelion on wallet_tests so that transactions do not enter stempool and effects can be seen immediately

Commit:
Fix: Remove 'test' statement

SmartArray committed:
Merge pull request #80 from barrystyle/feature/dandelion

Reimplement Dandelion Transactions

Commit:
cherry-picked "Litecoin: Disable RBF" (ltc-mweb/litecoin@0de5edb7e)

Commit:
Functional tests: Exported blockversion constant to external file

Commit:
Refactor: Removed dead code

Commit:
Fix: Removed duplicate lock statement

Commit:
Fix: Dandelion mutex LOCKs: added where required, removed where not

Commit:
Fix: Dandelion mutex LOCKs: added where required, removed where not

Commit:
Functional: Exported blockversion to external module

Commit:
Fix-Functional: adapted wallet_basic.py to DigiByte Wallet Parameters

Commit:
Fix-Functional: Adapted calls to generate()

Commit:
Fix-Functional: Adapted keypath to DigiByte Params

Commit:
Fix-Functional: Removed dandelion=0 argument from node config, as test_node.py will have this set by default

Commit:
Fix-Functional: wallet_backup.py: adapted to DigiByte block maturity parameters

Commit:
Fix-Functional: Added -easypow argument for regtest networks.
This will be needed to postpone difficulty adjustment modifications to a block in the future throughout the functional tests

Commit:
Fix-Functional: Fixed address specific tests (address.py)

Commit:
Fix-Functional: Install digibyte_scrypt for functional tests

Commit:
Fix-Functional: Added ENABLE_PYTHON_SCRYPT variable to travis conf

Commit:
Fix-Functional: feature_block.py: Adapted to DigiByte Blockchain Parameters

Warning: Mempool Resurrection doesn't work and needs to be checked.

Commit:
remove unnecessary hardfork code, as version MIN_PEER_PROTO_VERSION
is sufficient to prevent communications with older clients

gto90 committed:
Merge pull request #82 from barrystyle/feature/disable-rbf125

Disable RBF transaction signalling

gto90 committed:
Merge pull request #86 from barrystyle/feature/remove-legacy-hardfork

Remove legacy odocrypt fork code

digicontributer committed:
Merge pull request #84 from SmartArray/feature/fix-dandelion-locks

Fix: Dandelion mutex LOCK usages

Commit:
fix: mempool-resurrection: fixed logic error and invalid reorg condition

Commit:
Fix: wallet_reorgsrestore.py: ensure min relay fee is set

Commit:
fix: wallet_orphanedreward.py: fixed block yield parameters, cleaned up code

Commit:
fix: wallet_multiwallet.py: adjusted block yield amounts and adapted sqlite warning message in DigiByte Core

Commit:
fix: feature_assumevalid.py: adapted test parameters and expectations, CORE: increased peer timeouts

Commit:
fix: feature_config_args.py: fixed non-existing config check

Commit:
fix: feature_dersig.py: adjusted version parameters

Commit:
fix: dandelion: do not reuse node's fee estimator to prevent 'tx already being tracked' error

Commit:
fix: functional: added original MATURITY param (100)

Commit:
fix: feature_loadblock.py: use original COINBASE_MATURITY parameter

Commit:
fix: functional: replaced testnet addresses with corresponding DGB addresses

Commit:
fix: feature_notifications.py: updated testnet seed

Commit:
fix: feature_rbf.py: adjusted fee model

Commit:
fix: test_functional: address encodings fixed and imported hash160

Commit:
fix: functional: updated WIF testnet prefix

Commit:
fix: feature_segwit.py: fixed segwit tests by adjusting the key params

Commit:
fix: functional: updated fees used in taproot tests

Commit:
fix: feature_utxo_set_hash.py: updated utxo hashes

Commit:
fix: easypow: easypow parameter introduced for regtest

Commit:
refactor: pow.cpp: replaced tabs with spaces

Commit:
fix: functional: feature_versionbits_warning.py: adapted to digibyte regtest network params

Commit:
fix: output all difficulties in the getinfo response

Commit:
fix: functional: interface_digibyte_cli.py: adjusted block and multialgo parameters

Commit:
fix: functional: interface_rest.py: digibyte address and params introduced

Commit:
fix: primitives/transaction.h: bumped default transaction version to get support for BIP68

Commit:
fix: functional: mempool_accept: fixed fee and amounts

Commit:
fix: functional: mempool_accept_wtxid.py: adjusted fees and removed future code

Commit:
fix: functional: mempool_limit.py: adjusted fees

Commit:
fix: functional: use -easypow by default

Commit:
fix: functional: added max-fee param to chain_transaction

Commit:
fix: functional: mempool_package_onemore.py: adapted fees by default

Commit:
fix: functional: mempool_persist.py: adapted according to Digibyte network parameters

Commit:
fix: wallet_listsinceblock.py: fixed fees and amounts

Commit:
fix: wallet_listreceivedby.py: updated complete test

Commit:
fix: wallet_listdescriptors.py: updated test and test vectors

Commit:
fix: wallet_labels.py: fixed test and updated test vectors

Commit:
fix: functional: wallet_labels.py: updated test vectors (descriptors)

Commit:
fix: functional: wallet_keypool.py: updated tests and fees

Commit:
fix: functional: wallet_importmulti.py: updated test vectors

Commit:
fix: functional: wallet_importdescriptors.py: updated test vectors

Commit:
fix: functional: wallet_groups.py: fixed fee values

Commit:
fix: functional: wallet_groups.py: adjusted fee params for descriptors

Commit:
fix: functional: wallet_dump.py: adapted address prefixes

Commit:
fix: functional: wallet_disable.py: fixed address test vectors

Commit:
fix: functional: wallet_descriptor.py: adapted test conditions to three address types

Commit:
fix: functional: calculate correct coinbase amounts by default

Commit:
fix: functional: adapted genesis timestamp

Commit:
fix: functional: wallet_coinbase_category.py: fixed COINBASE_MATURITY param

Commit:
fix: functional: wallet_basic.py: respect DigiByte's lower COINBASE_MATURITY parameter

Commit:
fix: functional: wallet_balance.py: adapted tests to DigiByte's block emission

Commit:
fix: functional: wallet_backup.py: changed positional arg to named arg

Commit:
fix: functional: wallet_address_types.py: generate more coins by using COINBASE_MATURITY_ORIGINAL to have tests succeeding

Commit:
fix: functional: wallet_address_types.py: disable bech32m until implemented

Commit:
fix: functional: wallet_abondonconflict.py: respect min relay fee

Commit:
fix: functional: tool_wallet.py: fixed test by occurrences of bitcoin with digibyte

Commit:
fix: functional: rpc_txoutproof.py: fixed block count assumption

Commit:
fix: functional: rpc_signrawtransaction.py: fixed fees, adjusted activation times, replaced test vector for signing test

Commit:
fix: functional: rpc_signmessage.py: updated test vector

Commit:
fix: functional: rpc_scantxoutset.py: replaced test vectors

Commit:
fix: functional: rpc_rawtransaction.py: fixed emissions and fees

Commit:
xxxx

Commit:
fix: functional: rpc_psbt.py: modified fees and psbts

Commit:
fix: functional: rpc_packages.py: fixed fees and amounts

Commit:
fix: functional: rpc_net.py: fixed digibyte specific network parameters

Commit:
fix: functional: rpc_misc.py: removed future code

Commit:
fix: functional: rpc_invalidateblock.py: fixed checksum in address.py

Commit:
fix: functional: rpc_invalid_address_message.py: fixed test vectors

Commit:
fix: functional: rpc_getblockstats.py: generated new test vectors

Commit:
fix: functional: rpc_generateblock.py: fixed bech32 and legacy test vectors

Commit:
fix: functional: rpc_generate.py: fix output test vector

Commit:
fix: https://github.com/bitcoin/bitcoin/issues/22670

Commit:
fix: rounding issues: https://github.com/bitcoin/bitcoin/pull/22949

Commit:
fix: functional: rpc_fundrawtransaction.py: adjusted to DigiByte Core

Commit:
fix: functional: rpc_dumptxoutset.py: updated test vectors

Commit:
fix: functional: rpc_deriveaddresses.py: updated address test vectors

Commit:
fix: functional: rpc_decodescript.py: removed future code

Commit:
fix: functional: rpc_createmultisig.py: updated test vectors and fee

Commit:
fix: functional: rpc_blockchain.py: fixed tests by adjusting to current version 8.22

Commit:
fix: functional: rpc_addresses_deprecation.py: fixed call to generateblock() and adjusted test vectors

Commit:
fix: functional: p2p_invalid_messages.py: adjusted digibyte network p2p rules

Commit:
fix: functional: p2p_invalid_block.py: fixed calls to generateaddress() and bumped MATURITY test

Commit:
fix: functional: p2p_ibd_txrelay.py: adjusted max fee filter

Commit:
fix: functional: p2p_filter.py: fixed call to generate()

Commit:
fix: functional: p2p_compactblocks.py: adjusted fees

Commit:
fix: functional: p2p_blocksonly.py: adapted test to current DigiByte Core protocol

Commit:
fix: functional: mining_basic.py: fixed mining tests by adjusting block count and respecting algorithm

Commit:
fix: functional: mempool_reorg.py: respect DigiByte's COINBASE_MATURITY and use test_framework's newer functions

Commit:
fix: functional: mempool_spend_coinbase.py: respect DigiBytes smaller COIN_MATURITY parameter

Commit:
fix: functional: wallet_keypool.py: rounding error

Commit:
fix: functional: feature_fee_estimation.py: updated fees and backported tests

Commit:
added license header

Commit:
fix: functional: feature_proxy.py: backported test

Commit:
fix: functional: feature_rbf.py: backported test, fixed fees and amounts

Commit:
fix: functional: feature_utxo_set_hash.py: fixed hash test vectors

Commit:
fix: functional: mempool_packages.py: several improvements and fixes

Commit:
fix: functional: respect DigiByte's testnet4

Commit:
fix: testnet: removed invalid duplicate of powLimit

Commit:
testnet: added checkpoint

Commit:
testnet: selected block 546 as testnet checkpoint hash

Commit:
fix: functional: p2p_dos_header_tree.py: updated testnet vector

Commit:
fix: functional: p2p_feefilter.py: respect common version of clients for wtxid broadcast

Commit:
fix: functional: p2p_leak.py: bumped functional test client version to support wtxid relay

Commit:
functional: license header altered

Commit:
fix: functional: mempool_unbroadcast.py: updated fees

Commit:
fix: dandelion and broken unbroadcast (non-final fix, dandelion needs functional test)

Commit:
fix: functional: p2p_segwit.py: fixed fees, address prefix and amounts

Commit:
fix: functional: p2p_sendheaders.py: fixed digibyte-specific block header response expectations. 32 blocks sent per transit, instead of 16 in bitcoin

Commit:
fix: functional: p2p_tx_download.py: updated subsidy amounts and fees. Test working now due to dandelion bugfix

Commit:
fix: functional: rpc_generateblock.py: fixed call to MiniWallet's utxo scan

Commit:
fix: functional: rpc_packages.py: fixed calls and amounts, added code from upstream

Commit:
fix: functional: wallet_bumpfee.py: fixed fees, amounts and DigiByte specific maturities

Commit:
fix: functional: wallet_send.py: fixed DigiByte specific fee rates and amounts, removed future code

Commit:
fix: functional: wallet_signer.py: removed future code

Commit:
fix: functional: wallet_txn_doublespend.py: updated fees and amounts

Commit:
fix: functional: wallet_taproot.py: fixed fees and maturities

Commit:
fix: functional: feature_taproot.py: backported tests and updated fees

Commit:
fix: functional: mempool_packages.py: use tx hash instead of txid due to bumped protocol version to 70018

Commit:
fix: functional: deleted old files and commited all other changes

Commit:
fix: core: critical mempool bug fixed. All transactions entered the mempool, even if they were not supposed to

Commit:
fix: core: bumped protocol version so that wtxid relay becomes active

Commit:
fix: core: bumped high fee warning threshold to 1 DGB

Commit:
fix: functional: p2p_addr_relay.py: modified mocktime and removed initial getaddr broadcast

Commit:
fix: increased min relay fee of DigiByte

Commit:
fix: functional: wallet_basic.py: removed print instruction

Commit:
fix: functional: wallet_basic.py: increased factor of chainlimit due to DigiByte's small maturity param

Commit:
fix: functional: wallet_listtransactions.py: backported tests

Commit:
fix: functional: mempool_accept.py: backported test

Commit:
fix: functional: feature_rbf.py: removed obsolete argument

Commit:
fix: functional: p2p_leak.py: updated required common version for BIP155

Commit:
fix: core: net_processing.cpp: bumped required version for addrv2 protocol consensus from 70016 to 70018

Commit:
fix: functional: wallet_basic.py: introduced multiplicator for long chain tx required due to small maturity of DigiByte coins

Commit:
fix: functional: wallet_importdescriptors.py: fixed amounts to ensure that only one output is created

Commit:
fix: functional: wallet_send.py: fixed address prefix check

Commit:
fix: functional: p2p_dandelion.py: repaired python script to be up-to-date with test_framework.
2/3 subtests are failing. Core code needs to be adjusted.

Commit:
fix: functional: temporarily disabled signet and dandelion tests

Commit:
fix: functional: feature_fee_estimation.py: adjusted fee

Commit:
Merge branch 'fix/core-bugs' into test/merge

Commit:
Merge branch 'fix/functional' into test/merge

SmartArray committed:
Fixed linking error

Previous change to Makefile.am caused Linking issue on Debian based systems.

gto90 committed:
Merge pull request #89 from SmartArray/fix/core-functional

Core Improvements and Bugfixes

ycagel committed:
Merge pull request #90 from SmartArray/fix/functional

Fix: Functional Tests

gto90 committed:
Added a step to install the missing pypandoc dependency.

gto90 committed:
Added missing pypandoc dependency installation

gto90 committed:
Added missing workflow steps to install pypandoc

gto90 committed:
Added pandoc installation step

gto90 committed:
Removing extra mac pandoc installation step

gto90 committed:
Explicityly specifying macOS-11 for the action runner

gto90 committed:
Falling back to macos-10.15 for the github runner.

gto90 committed:
Trying macos-12 for the github runner

gto90 committed:
Aligning mac configure step with linux configure step

gto90 committed:
Removed commented out workflow steps

gto90 committed:
Troubleshooting Boost Errors

gto90 committed:
Adding Boost linking step

gto90 committed:
Added step to archive test-suite.log as an artifact

gto90 committed:
Troubleshooting test artifact upload

gto90 committed:
Troubleshooting test suite log artifact uploading

gto90 committed:
Removing directory list step

Commit:
Updated regtest hash

Commit:
Updated fee to match minRelayFee

Commit:
Updated Regtest UtxoHash due to updated minRelayFee

Commit:
fix unit tests: TX default version bumped to v2

Commit:
Removed return statement of validation_tests.cpp

gto90 committed:
fix: Updated GitHub Actions to Install Berkley DB 4.8 and remove the `—with-incompatible-bdb` flag.

Signed-off-by: GTO90 <gto90@protonmail.com>

gto90 committed:
fix: Updated GitHub Actions to remove manual installation of BerkleyDB

Signed-off-by: GTO90 <gto90@protonmail.com>

gto90 committed:
fix: Attempting to enable BDB cxx headers

Signed-off-by: GTO90 <gto90@protonmail.com>

gto90 committed:
fix: Attempting to fix BerkleyDB 4.8 installation

Signed-off-by: GTO90 <gto90@protonmail.com>

gto90 committed:
fix: Attempting to fix BerkleyDB installation

gto90 committed:
fix: Attempt to fix BDB4 installation

Signed-off-by: GTO90 <gto90@protonmail.com>

gto90 committed:
fix: BDB Configuration

Signed-off-by: GTO90 <gto90@protonmail.com>

gto90 committed:
fix: Attempting BDB fix using Bitcoin repo

Signed-off-by: GTO90 <gto90@protonmail.com>

gto90 committed:
Revert "fix: Attempting BDB fix using Bitcoin repo"

This reverts commit 1e0630c12785644737c3cc35b907dae094c2f0fa.

Signed-off-by: GTO90 <gto90@protonmail.com>

gto90 committed:
fix: Cleanup GitHub Actions

Signed-off-by: GTO90 <gto90@protonmail.com>

gto90 committed:
Debugging Linux GitHub CI Action

gto90 committed:
Debugging GitHub Action

gto90 committed:
debugging GitHub Action

gto90 committed:
Debugging GitHub Actions

gto90 committed:
Debugging GitHub Actions

gto90 committed:
Disabling tests and gui on the `ci-coverage-ubuntu-no-tests-no-qt` action

JaredTate committed:
Add Sudo

Permissions

JaredTate committed:
Add Sudo

Permissions

Commit:
Fix SendCoins - We no longer have an RBF dialogue

Commit:
build: patch qt to explicitly define previously implicit header include

Commit:
build: update mac full qt github action

JaredTate committed:
Bump Min Relay Fee Functional Tests

JaredTate committed:
Bump Fee Span

JaredTate committed:
Merge pull request #93 from DigiByte-Core/fix/unit_tests

Update Unit Test Vectors

JaredTate committed:
Disable Interface ZMQ Test Temporarily

JaredTate committed:
Add Functional Tests Github Runner

JaredTate committed:
Remove Pull Request Trigger so Checks Dont Run Twice

JaredTate committed:
Remove Functional Tests

ycagel committed:
Merge pull request #94 from DigiByte-Core/fix/functional_tests

WIP: Fix Functional Tests

gto90 committed:
Validating Depends Actions

JaredTate committed:
Subsidy Regtest

JaredTate committed:
Boost 1.77 Fix

Boost 1.77 Fix

JaredTate committed:
Revert "Subsidy Regtest"

This reverts commit 5d8beb7f6aab7ebd8e7c2e7e3ba77339d0cecf42.

JaredTate committed:
Make Deploy Working

Need to make sure local Python version matches this for Make Deploy to build the dmg

Commit:
fix(functional): wallet_multiwallet.py: adapted assertion to support multiple boost versions

Commit:
fix(functional): p2p_addr_relay.py and wallet_resendwallettransactions.py: -easypow parameter added for regtest transactions

Commit:
fix(walletdir): removing trailing path separator from walletdir path, as tested in init_tests.cpp by walletinit_verify_walletdir_no_trailing2 and walletinit_verify_walletdir_no_trailing

Commit:
feat(functional): display SKIP message after failing. Corresponding to https://github.com/bitcoin/bitcoin/pull/26031

Commit:
fix(dandelion): prevent notifications when transactions are added to the stempool

Commit:
fix(functional): zmq tests will now succeed. ipv6 will not tested until 8.23

Commit:
fix(functional): deleted old unused test `rpc_zmq.py` (as in https://github.com/bitcoin/bitcoin/commit/42a995ae4819ea472b211eb5088727a422fcbeb8)

gto90 commented:
@SmartArray , when you get a free moment, can you retarget this pr to here: https://github.com/DigiByte-Core/digibyte/tree/feature/8.22.0

JaredTate committed:
Disable SSL Tests

Bitcoin Core completely ripped out SSL tests as OpenSSL3 changed a ton of stuff and broke many things in core tests.
